src/compile/common.ts

```diff
@@ -183,3 +187,52 @@ export function timeFormat(model: Model, channel: Channel): string {
   const fieldDef = model.fieldDef(channel);
   return timeFormatExpr(fieldDef.timeUnit, isAbbreviated(model, channel, fieldDef));
 }
+
+export function hasGeoTransform(model: UnitModel): boolean {
+  const geoPathFieldDef = model.encoding().geopath;
+  if (model.mark() === PATHMARK) {
+    if (geoPathFieldDef && geoPathFieldDef.type === GEOJSON) {
+      // If the mark is path and geopath encoding is defined with geojson type,
+      // it means the user wants to have geopath transform.
+      return true;
+    }
+  }
+  // For geo transform, we only plot latitude/longitude against
+  // x/y because it would not make sense for other measures like color.
+  return containsLatLong(model.encoding());
+}
+
+export function geoTransform(model: UnitModel) {
+  const translate = model.projection().translate;
+  const scale     = model.projection().scale;
+  const center    = model.projection().center;
+  const rotate    = model.projection().rotate;
+  const precision = model.projection().precision;
+  const clipAngle = model.projection().clipAngle;
+  let spec = {};
+  if (model.mark() === PATHMARK) {
+    // return geoPath transform
+    const geoPathFieldDef = model.encoding().geopath;
+    spec = { type: 'geopath', field: geoPathFieldDef.field };
+  } else {
+    spec = { type: 'geo' };
+    const xFieldDef = model.encoding().x;
+    const yFieldDef = model.encoding().y;
+    spec = extend(spec,
+        xFieldDef.type === LATITUDE ? { lat : xFieldDef.field }
+            : xFieldDef.type === LONGITUDE ? { lon : xFieldDef.field }
+            : {});
+    spec = extend(spec,
+    yFieldDef.type === LATITUDE ? { lat : yFieldDef.field }
+        : yFieldDef.type === LONGITUDE ? { lon : yFieldDef.field }
+        : {});
+  }
+    spec = extend(spec, { projection : model.projection().type});
+    spec = extend(spec, translate !== undefined ? { translate : translate} : {});
```

Comment:
if we use `let` anyway, let's do:

```js
if(translate !== undefined) { 
   spec.translate = translate;
}
```

(Same for the rest)

(You can also be lazy and do for loop)

```js
['translate', ...].forEach((prop) => {
   if (projection[prop] !== undefined) {
      spec[prop] = projection[prop]
   }
}
```